gcs/private/util.bzl

```
@@ -78,8 +83,14 @@ def download_and_extract_args(attr, bucket_name, remote_path):
         args.pop("rename_files")
     return args
 
-def bucket_url(bucket, object_path):
-    return "https://storage.googleapis.com/{}/{}".format(bucket, url_encode(object_path))
+# Use the JSON API, which allows for more options https://cloud.google.com/storage/docs/json_api/v1/objects/get
+def bucket_url(bucket, object_path, generation):
+    return "https://storage.googleapis.com/storage/v1/b/{}/o/{}?alt=media".format(
```

```suggestion
    return "https://storage.googleapis.com/storage/v1/b/{}/o/?alt=media{}".format(
```

I think for a valid url query string, we always want the "?" first, followed by "&"-joined kv pairs.

---

This is wrong. Sorry for that.